package.json

```diff
@@ -23,6 +23,7 @@
     "requireindex": "^1.2.0"
   },
   "devDependencies": {
+    "@typescript-eslint/parser": "^4.19.0",
```

Comment:
Do we really, strongly need this dependency?

I think I would slightly prefer we handle this the same way as ESLint does: By creating fixture parsers which just return the exact AST that we expect from the real typescript-eslint parser.

Removing the dependency would slightly insulate us from any accidental breaking changes.

Here's an example from the ESLint repo: https://github.com/eslint/eslint/blob/6f4abe5d5ade2711cc4c21bc8485af952763c2d3/tests/lib/rules/require-await.js#L49

Reply:
I'd be happy to switch to the fixture approach if you prefer, although executing the actual TS parsing code for these new test cases so that we can find out about breaking changes during version updates (and thus handle them properly) also sounds like an upside.

Reply:
Do you know of a good way to generate the AST for the fixture? I tried both of these:

* taking the JSON output from astexplorer.net
* running the parse() function manually

  ```
  const parser = require('@typescript-eslint/parser');
  const result = parser.parse('QUnit.module("module", function(this: LocalTestContext, hooks) {hooks.afterEach(function() {}); });',{ range:true });
  console.log(JSON.stringify(result))
  ```

Then I had to add this to the AST:

```
"comments": [],
"tokens": []
```

but it still gave me this error when running the test case with the custom fixture parse function:

```
A fatal parsing error occurred: Parsing error: AST is missing location information.
```

Reply:
Let me think on this some more. I'm not completely convinced but I'm more open to the idea of having the devDependency than I was. If I decide I'm strongly against it I'll look into your question as well, sound good? Thanks for digging in so far.